Server/src/redis/MsgBusImpl_redis.cpp

```diff
@@ -0,0 +1,243 @@
+#include <cstdlib>
+#include <cstring>
+#include <iostream>
+
+#include <cinttypes>
+
+#include <netdb.h>
+#include <unistd.h>
+
+#include <thread>
+#include <arpa/inet.h>
+
+#include "MsgBusImpl_redis.h"
+#include "RedisManager.h"
+
+
+
+
+
+using namespace std;
+
+/******************************************************************//**
+ * \brief This function will initialize and connect to Kafka.
+ *
+ * \details It is expected that this class will start off with a new connection.
+ *
+ *  \param [in] logPtr      Pointer to Logger instance
+ *  \param [in] cfg         Pointer to the config instance
+ ********************************************************************/
+MsgBusImpl_redis::MsgBusImpl_redis(Logger *logPtr, Config *cfg, BMPListener::ClientInfo *client) {
+    logger = logPtr;
+    this->cfg = cfg;
+    redisMgr_.Setup(logPtr, client);
+    std::vector<std::string> tables;
+    tables.emplace_back(BMP_CFG_TABLE_NEI);
+    tables.emplace_back(BMP_CFG_TABLE_RIB_IN);
+    tables.emplace_back(BMP_CFG_TABLE_RIB_OUT);
+
+    redisMgr_.ReadBMPTable(tables);
+}
+
+/**
+ * Destructor
+ */
+MsgBusImpl_redis::~MsgBusImpl_redis() {
+    redisMgr_.ExitRedisManager();
+}
+
+
+/**
+ * Abstract method Implementation - See MsgBusInterface.hpp for details
+ */
+void MsgBusImpl_redis::update_Peer(obj_bgp_peer &peer, obj_peer_up_event *up, obj_peer_down_event *down, peer_action_code code) {
+
+    // Below attributes will be populated if exists, and no matter bgp neighbor is up or down
+    std::vector<swss::FieldValueTuple> fieldValues;
+    fieldValues.reserve(30);
+    std::vector<std::string> keys;
+    keys.emplace_back(peer.peer_addr);
+
+    fieldValues.emplace_back(std::make_pair("peer_addr", peer.peer_addr));
```

Comment:
>std:: [](http://example.com/codeflow?start=29&length=5)

you can simplify by just remove `std::`.

Reply:
fixed